### ISSUE TITLE
Some instantiations of rdf/3 give incorrect results

### ISSUE THREAD

**Issue by wouterbeek** [SWI-Prolog/packages-semweb#78]
When the same variable was used in the object term position, as well as in the subject and/or predicate term position, object term bindings from the database would not be considered at all.

For example:

```prolog
?- [library(semweb/rdf11)].
?- rdf_assert(s, p, o).
?- rdf(X, Y, Y).
X = s,
Y = p.
```

**Comment by likelion** [comment]
The problem is somewhere in the native code. Seems to be connected to object:

```prolog
?- rdf(X, Y, X).
X = s,
Y = p.
```

**Comment by likelion** [comment]
Fixed in #95.

**Comment by wouterbeek** [comment]
Thanks!
